ext/libclementine-tagreader/tagreader.cpp

```diff
@@ -56,13 +56,14 @@
 
 #include <sys/stat.h>
 
-#include "fmpsparser.h"
 #include "core/logging.h"
 #include "core/messagehandler.h"
 #include "core/timeconstants.h"
+#include "fmpsparser.h"
+#include "gmereader.h"
 
 // Taglib added support for FLAC pictures in 1.7.0
-#if (TAGLIB_MAJOR_VERSION > 1) || \
+#if (TAGLIB_MAJOR_VERSION > 1) ||                                              \
```

Comment:
Please revert all the whitespace changes.

Reply:
Hey thanks I'll get on those. I ran Clang-format but it seems to have done the opposite of how I wanted it. Do you happen to have a clang-format for clementine?

I'll get onto the other changes tomorrow.

Reply:
You should be able to just run `make format` or (experimental) you can follow the link clang-formatter posted and sign in there to commit it for you.

Reply:
So, for the formatting I've been having some issues. With ```make format``` I've been getting an error from urllib2 that says ```HTTP ERROR 405: Method Not Allowed```. Error log looks as follows.

```
Traceback (most recent call last):
File "/home/eoin/Source/Clementine/dist/format.py", line 84, in <module>
main()
File "/home/eoin/Source/Clementine/dist/format.py", line 59, in main
response = urllib2.urlopen(args.url, original)
File "/usr/lib/python2.7/urllib2.py", line 154, in urlopen
return opener.open(url, data, timeout)
File "/usr/lib/python2.7/urllib2.py", line 435, in open
response = meth(req, response)
File "/usr/lib/python2.7/urllib2.py", line 548, in http_response
'http', request, response, code, msg, hdrs)
File "/usr/lib/python2.7/urllib2.py", line 467, in error
result = self._call_chain(*args)
File "/usr/lib/python2.7/urllib2.py", line 407, in _call_chain
result = func(*args)
File "/usr/lib/python2.7/urllib2.py", line 654, in http_error_302
return self.parent.open(new, timeout=req.timeout)
File "/usr/lib/python2.7/urllib2.py", line 435, in open
response = meth(req, response)
File "/usr/lib/python2.7/urllib2.py", line 548, in http_response
'http', request, response, code, msg, hdrs)
File "/usr/lib/python2.7/urllib2.py", line 473, in error
return self._call_chain(*args)
File "/usr/lib/python2.7/urllib2.py", line 407, in _call_chain
result = func(*args)
File "/usr/lib/python2.7/urllib2.py", line 556, in http_error_default
raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
urllib2.HTTPError: HTTP Error 405: Method Not Allowed
CMakeFiles/format.dir/build.make:57: recipe for target 'CMakeFiles/format' failed
make[3]: *** [CMakeFiles/format] Error 1
CMakeFiles/Makefile2:163: recipe for target 'CMakeFiles/format.dir/all' failed
make[2]: *** [CMakeFiles/format.dir/all] Error 2
CMakeFiles/Makefile2:170: recipe for target 'CMakeFiles/format.dir/rule' failed
make[1]: *** [CMakeFiles/format.dir/rule] Error 2
Makefile:203: recipe for target 'format' failed
make: *** [format] Error 2 
```

When I tried using the online clang-formater link, it gave me an error because It didn't know what to do with the .proto file. The online formatter error was as follows:

```
Failed to format file: No formatter for file: ext/libclementine-remote/remotecontrolmessages.proto
```

The two errors might actually be related to one another, but I'm not sure. For now, I'm going to try downloading the diff file and applying it to my code manually to fix formatting issues. Thanks for the help @hatstand and I'm sorry for my inexperience when it comes to the whole clang formatting thing. 

As for the ACTUAL changes, it seems that ASCII works as intended (I was unsure because of the format originally) so that has been changed wherever necessary. I also tried to be more safe about array indexing (because, you're right, the array return size is not guaranteed.)

Reply:
I never got the make format to work either. I just manually apply the diff provided by clang-formatter as a patch.